scripts/utils.py

```diff
@@ -0,0 +1,134 @@
+#! /usr/bin/env python3
+
+# Licensed Materials - Property of IBM
+# 57XX-XXX
+# (c) Copyright IBM Corp. 2021
+
+from enum import Enum
+import json
+import os
+import subprocess
+import sys
+from typing import List
+from unittest import result
+
+from scripts.const import FILE_MAX_EXT_LENGTH, FILE_TARGET_MAPPING
+
+class Colors(str, Enum):
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+
+def colored(message: str, color: Colors) -> str:
+    """Returns a colored message if supported
+    """
+    if support_color():
+        return f"{color}{message}{Colors.ENDC}"
+    else:
+        return f"{message}"
+
+
+def support_color():
+    return True
+
+def read_ibmi_json(path, parent_value):
+    if path.exists():
+        with path.open() as f:
+            data = json.load(f)
+            try:
+                objlib = parse_placeholder(data['build']['objlib'])
+                
+            except Exception:
+                objlib = parent_value[0]
+            try:
+                tgtCcsid = data['build']['tgtCcsid']
+            except Exception:
+                tgtCcsid = parent_value[1]
+            return (objlib, tgtCcsid)
+    else:
+        return parent_value
+
+def parse_placeholder(varName):
+    if varName.startswith("&") and len(varName) > 1:
+        varName = varName[1:]
+        try:
+            value = os.environ[varName]
+            return value
+        except Exception:
+            print(colored(f"{varName} must be defined first in the environment variable.", Colors.FAIL))
+            exit(1)
+    else:
+        return varName
+
+def read_iproj_json(iproj_json_path):
+    try:
+        with iproj_json_path.open() as f:
+            iproj_json = json.load(f)
+            objlib = parse_placeholder(iproj_json["objlib"]) if "objlib" in iproj_json else ""
```

Comment:
are we pulling in defaults if these are not specified?  i.e. *CURLIB and *CRTDFT